src/bootstrap/test.rs

```diff
@@ -827,15 +827,28 @@ impl Step for RustdocGUI {
             }
 
             let out_dir = builder.test_out(self.target).join("rustdoc-gui");
-            let mut command = builder.rustdoc_cmd(self.compiler);
-            command.arg("src/test/rustdoc-gui/lib.rs").arg("-o").arg(&out_dir);
-            builder.run(&mut command);
 
+            // We remove existing folder to be sure there won't be artifacts remaining.
+            let _ = fs::remove_dir_all(&out_dir);
+
+            // We generate docs for the libraries present in the rustdoc-gui's src folder.
+            let libs_dir = Path::new("src/test/rustdoc-gui/src");
```

Comment:
This needs to be joined onto `builder.config.src`, otherwise this won't work when invoked from a different working directory.